src/cmake/dependency_utils.cmake

```diff
@@ -0,0 +1,422 @@
+# Copyright Contributors to the OpenImageIO project.
+# SPDX-License-Identifier: Apache-2.0
+# https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+
+set_cache (${PROJECT_NAME}_REQUIRED_DEPS ""
+           "Additional dependencies to consider required (semicolon-separated list, or ALL)")
+set_cache (${PROJECT_NAME}_OPTIONAL_DEPS ""
+           "Additional dependencies to consider optional (semicolon-separated list, or ALL)")
+set_option (${PROJECT_NAME}_ALWAYS_PREFER_CONFIG
+            "Prefer a dependency's exported config file if it's available" OFF)
+set_cache (${PROJECT_NAME}_DEPENDENCY_BUILD_TYPE "Release"
+           "Build type for locally built dependencies")
+
+if (MSVC)
+    # I haven't been able to get Windows local dependency builds to work with
```

Comment:
What problems did you run into while building static libs on Windows? Was it not worth the effort - significantly complicating the process? Is it related to symbol visibility?

Reply:
I hesitate to give much detail here, because my recollection is incomplete, and any problems I had are likely due to my lack of experience with Windows' quirks. I feel confident that somebody with real Windows development experience could probably get it working with static libs rather quickly.